edn-react/src/services/AuthFetch.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from "react";
+import { useAuth } from "@clerk/clerk-react";
+
+export const useAuthFetchData = <T>(url: string) => {
```

Comment:
I have a completely different service to handle fetching data when the call needs to be authenticated. This is mainly because the Clerk sdk gives back a promise for the auth token and causes re-render issues when trying to read from it. I had to put in a flag to stop the component that calls this from constantly rendering every time the token is updated after load and didn't want that to be in every fetch call. 

This should probably be refactored later to let us just skip grabbing the auth token if we don't need it and combine it into one hook.